hyperledger/firefly
# Implement API add namespace route

## Description 
This PR implements the SPI route for adding new namespaces with the appropriate plugins.
I saw that this implementation was missing, even though the [description](https://github.com/hyperledger/firefly/blob/2fbd5b68c09dc4fb00746c93b1c3a5cd9e8b18f3/internal/coremsgs/en_api_translations.go#L175) is already created, but it is describing the wrong [route](https://github.com/hyperledger/firefly/blob/2fbd5b68c09dc4fb00746c93b1c3a5cd9e8b18f3/internal/apiserver/route_spi_get_namespace_by_name.go#LL35C28-L35C56).

Since there is already a method that dynamically reloads the config file, I've used the same mechanism to reload it when the content is changed after this route call.     
It seams to me that this is the easy and clean way to implement this feature. Please, do let me know if this should be done in another, better way.

## Manual tests
Run the binary or a container, open the SPI Swagger interface and you should see the new POST route `spi#/Global/spiPostAddNamespaces`. The POST body should be as the following example: 
```
{
  "defaultKey": "0x36acbe2c5e1961552fd27f4e93723f3e8f6e778d",
  "description": "New customer",
  "name": "customer1",
  "plugins": [
    "database0",
    "blockchain0",
    "erc20_erc721"
  ]
}
```
After the call, run another GET call to verify that the new namespace has been successfully created.

## Comment
The start up that I'm working with, really needs this feature, so for now we're working of this branch. 
We have a use case where each new namespace should run on a dedicated private chain.
But we're hoping that this will be adopted into the main codebase so that we don't need to maintain our own fork.
